File: assets/js/last_reports.js
// Images taken from https://www.insectimages.org/index.cfm

const parseHTML = (data) => {
    let html = "";
    data.forEach(element => {
        html = html.concat(`
            <div class="home-report">
                <div class="description">
                    <p><b>Fecha:</b> ${element.fecha}</p>
                    <p><b>Comuna:</b> ${element.comuna}</p>
                    <p><b>Sector:</b> ${element.sector}</p>
                    <p><b>Tipo:</b> ${element.tipo}</p>
                </div>
                <div class="image">
                    <img src="${element.foto}" alt="report-img">
                </div>
            </div>`);
    });
    return html
};

/* **** MAIN **** */

fetch('./cgi-bin/last_reports.py').then(response => {
    console.debug('CGI RESPONSE', response);
    return response.json();
}).then(data => {
    const result = data.Result;
    console.debug('LAST 5 ',data);
    let lastReports = document.getElementById('lastReports');
    lastReports.innerHTML = parseHTML(result);
}).catch(e => {
    console.error(e)
});

File: cgi-bin/last_reports.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-
import cgi
import cgitb;
cgitb.enable()
import db
import json
import os

reportDB = db.ReportDB()
print("Content-type:application/json\r\n\r\n")
response = {
    'Result': [{
        'fecha': str(element[0]),
        'comuna': element[1], 
        'sector': element[2],
        'tipo': element[3],
        'foto': element[4][0] + element[4][1] 
        } for element in reportDB.get_last_reports()]
}
print(json.JSONEncoder().encode(response))